File: Leetcode/1. Two Sum/twoSum.java
import java.util.Arrays;
import java.util.HashMap;

// https://leetcode.com/problems/two-sum/description/

class Solution {
  public static int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> hash = new HashMap<>();
    int[] ar = new int[2];

    for (int i = 0; i < nums.length; i++) {
      int value = target - nums[i];
      if(hash.containsKey(value)) {
        ar[0] = hash.get(value);
        ar[1] = i;
      }else {
        hash.put(nums[i], i);
      }
    }
    return ar;
  }

  public static void main(String[] args) {

    int [] ar = {2,7,11,15};
    System.out.println(Arrays.toString(twoSum(ar, 9)));
  }
}

File: Leetcode/1. Two Sum/twoSum.rb
# https://leetcode.com/problems/two-sum/description/

def two_sum(nums, target)

  hash = {}
  nums.each_with_index do |number, index|

    if hash[target - number]
        return [hash[target - number], index]
    else
        hash[number] = index
    end
  end
end

#print two_sum([2,7,11,15], 9) == [0, 1]
#print two_sum([3,2,4], 6) == [1,2]
#print two_sum([3,3], 6)
print two_sum([11, 8, 1, 7], 9)
